agent/nmap_agent.py

```diff
@@ -141,9 +142,24 @@ def _emit_network_scan_finding(self, scan_results: Dict[str, Any], normal_result
         scan_result_technical_detail = process_scans.get_technical_details(scan_results)
         if normal_results is not None:
             technical_detail = f'{scan_result_technical_detail}\n```xml\n{normal_results}\n```'
-            self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
-                                      technical_detail=technical_detail,
-                                      risk_rating=vuln_mixin.RiskRating.INFO)
+            port = scan_results.get('nmaprun', {}).get('host', {}).get('@portid', '')
+            res = scan_results.get('nmaprun', {}).get('host', {}).get('address', '')
+            if res.get('@addrtype', '') == 'ipv4':
+                self.report_vulnerability(entry=kb.KB.NETWORK_PORT_SCAN,
+                                          technical_detail=technical_detail,
+                                          risk_rating=vuln_mixin.RiskRating.INFO,
+                                          vulnerability_location=vuln_mixin.VulnerabilityLocation(
+                                              metadata=[],
+                                              asset=ipv4.IPv4(host=res.get('@addr', ''), mask=port)
+                                          ))
+            else:
```

Comment:
I don't see the case for the domain target ? In that case you should emit the hostname as a domain asset.